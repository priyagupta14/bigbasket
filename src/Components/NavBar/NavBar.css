.dark-navbar {
  display: flex;
  justify-content: space-between;
  background-color: black;
  padding: 1rem;
}
.light-navbar{
  display: flex;
  justify-content: space-between;
  background-color: rgb(156, 151, 151);
  padding: 1rem;
}
.left-side-navbar{
  display: flex;
}

