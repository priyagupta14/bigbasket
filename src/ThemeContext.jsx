import React from 'react';

export const theme = {
  light: {
    backgroundColor: 'white',
    color: 'black',
  },
  dark: {
    backgroundColor: 'black',
    color: 'white',
  },
};

// export const ThemeContext = React.createContext(theme.light);
export const ThemeContext = React.createContext('white');
export const ThemeButton = React.createContext('grey');
