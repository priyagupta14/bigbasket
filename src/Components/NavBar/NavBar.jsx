/* eslint-disable react/prop-types */
import React, { useContext } from 'react';
import './NavBar.css';
import { Link } from 'react-router-dom';
import { ThemeContext } from '../../ThemeContext';
import MyBasket from '../MyBasket/MyBasket';

const NavBar = (props) => {
  const theme = useContext(ThemeContext);
  return (
    <div className={theme === 'light' ? 'light-navbar' : 'dark-navbar'}>
      <Link style={{ textDecoration: 'none', color: 'white' }} to="/">
        <br />
        E-Shopper
      </Link>
      <div className="left-side-navbar">
        <Link
          style={{ textDecoration: 'none', color: 'white' }}
          to="/allOrders"
        >
          <br />
          All Orders
        </Link>
        <Link style={{ textDecoration: 'none', color: 'white' }} to="/cart">
          <p>MyBasket</p>
          <MyBasket itemInCart={props.itemInCart} />
        </Link>
      </div>
    </div>
  );
};
export default NavBar;
