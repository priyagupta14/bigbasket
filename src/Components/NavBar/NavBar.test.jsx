import React from 'react';
import {
  fireEvent, render, screen,
} from '@testing-library/react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import { ThemeContext } from '../../ThemeContext';

import NavBar from './NavBar';

describe(NavBar.name, () => {
  const itemCount = 0;
  it('should match default theme snapshot', () => {
    const { container } = render(
      <BrowserRouter>
        <ThemeContext.Provider value="light">
          <NavBar itemInCart={itemCount} />
        </ThemeContext.Provider>
      </BrowserRouter>,
    );
    expect(container).toMatchSnapshot();
  });
  it('should match light theme snapshot', () => {
    const { container } = render(<BrowserRouter><NavBar itemInCart={itemCount} /></BrowserRouter>);
    expect(container).toMatchSnapshot();
  });
  it('should take to e-shoppers page', () => {
    render(<BrowserRouter><NavBar itemInCart={itemCount} /></BrowserRouter>);
    const eShoppers = screen.getByText('E-Shopper');
    fireEvent.click(eShoppers);
    expect(document.location.href).toBe('http://localhost/');
  });
  it('should take to All Orders page', () => {
    render(<BrowserRouter><NavBar itemInCart={itemCount} /></BrowserRouter>);
    const allOrders = screen.getByText('All Orders');
    fireEvent.click(allOrders);
    expect(document.location.href).toBe('http://localhost/allOrders');
  });
  it('should take to Cart page', () => {
    render(<BrowserRouter><NavBar itemInCart={itemCount} /></BrowserRouter>);
    const cart = screen.getByText('MyBasket');
    fireEvent.click(cart);
    expect(document.location.href).toBe('http://localhost/cart');
  });
});
