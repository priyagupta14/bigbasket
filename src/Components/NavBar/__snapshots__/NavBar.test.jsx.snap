// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should match default theme snapshot 1`] = `
<div>
  <div
    class="light-navbar"
  >
    <a
      href="/"
      style="text-decoration: none; color: white;"
    >
      <br />
      E-Shopper
    </a>
    <div
      class="left-side-navbar"
    >
      <a
        href="/allOrders"
        style="text-decoration: none; color: white;"
      >
        <br />
        All Orders
      </a>
      <a
        href="/cart"
        style="text-decoration: none; color: white;"
      >
        <p>
          MyBasket
        </p>
        <div
          class="mybasket-container"
        >
          <p>
            mycart
            0
          </p>
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`NavBar should match light theme snapshot 1`] = `
<div>
  <div
    class="dark-navbar"
  >
    <a
      href="/"
      style="text-decoration: none; color: white;"
    >
      <br />
      E-Shopper
    </a>
    <div
      class="left-side-navbar"
    >
      <a
        href="/allOrders"
        style="text-decoration: none; color: white;"
      >
        <br />
        All Orders
      </a>
      <a
        href="/cart"
        style="text-decoration: none; color: white;"
      >
        <p>
          MyBasket
        </p>
        <div
          class="mybasket-container"
        >
          <p>
            mycart
            0
          </p>
        </div>
      </a>
    </div>
  </div>
</div>
`;
